File: spaceship_crm/java8/spaceship_crm_java8_jsp_jpa/src/main/java/com/spaceship/crm/entity/UserInfo.java
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.OneToOne;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class UserInfo {
    @Id
    @GeneratedValue
    private Long id;

    private String phone;
    private String address;

    @OneToOne
    private User user;

}

File: spaceship_crm/kotlin/spaceship_crm_kotlin_spring_mustache_jpa/src/main/kotlin/com/spaceship/crm/entity/UserInfo.kt
/**
 * Author veni.vidi.dev (veni.vidi.dev@gmail.com)
 */
package com.spaceship.crm.entity

import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id
import javax.persistence.OneToOne

@Entity
class UserInfo(
        @Id
        @GeneratedValue
        var id: Long,
        var phone: String? = null,
        var address: String? = null,

        @OneToOne
        var user: User? = null
)